layer4/routes_test.go

```diff
@@ -20,7 +20,7 @@ import (
 type testIoMatcher struct {
 }
 
-func (testIoMatcher) CaddyModule() caddy.ModuleInfo {
+func (*testIoMatcher) CaddyModule() caddy.ModuleInfo {
```

Comment:
Why the change to pointers?

Reply:
Otherwise I have a warning in the IDE saying the use of both value and pointer receivers for the same type is discouraged. Anyway, pointers don't make any harm there.